sno/merge_util.py

```diff
@@ -562,6 +563,16 @@ def output(self, output_format):
         elif output_format == "geojson":
             return geojson_row(self.feature, self.pk_field)
 
+    def matches_filter(self, conflict_filter):
+        if conflict_filter == UNFILTERED:
+            return True
+        dataset_filter = conflict_filter.get(self.dataset_path, None)
+        if not dataset_filter:
+            return False
+        return dataset_filter == UNFILTERED or (
+            not self.is_meta and str(self.pk) in dataset_filter
```

Comment:
```suggestion
            str(self.pk) in dataset_filter and not self.is_meta
```
(same thing, just clearer precedence)

Reply:
Not done - `self.pk` fails if `self.is_meta`

To make it a bit more readable, I added self.is_feature that is just the negation of self.is_meta